ethereum/forge-test/GovernanceStructs.t.sol

```diff
@@ -0,0 +1,320 @@
+// test/Messages.sol
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "../contracts/GovernanceStructs.sol";
+import "forge-test/rv-helpers/TestUtils.sol";
+
+contract TestGovernanceStructs is TestUtils {
+    GovernanceStructs gs;
+
+    function setUp() public {
+        gs = new GovernanceStructs();
+    }
+
+    function testParseContractUpgrade(
+        bytes32 module,
+        uint16 chain,
+        bytes32 newContract
+    ) public {
+        uint8 action = 1;
+
+        bytes memory encodedUpgrade = abi.encodePacked(
+            module,
+            action,
+            chain,
+            newContract
+        );
+
+        assertEq(encodedUpgrade.length, 67);
+        
+        GovernanceStructs.ContractUpgrade memory cu =
+            gs.parseContractUpgrade(encodedUpgrade);
+
+        assertEq(cu.module, module);
+        assertEq(cu.action, action);
+        assertEq(cu.chain, chain);
+        assertEq(cu.newContract, address(uint160(uint256(newContract))));
+    }
+
+    function testParseContractUpgrade_KEVM(
+        bytes32 module,
+        uint16 chain,
+        bytes32 newContract
+    ) public symbolic(address(gs)) {
+        testParseContractUpgrade(module, chain, newContract);
+    }
+
+    function testParseContractUpgradeWrongAction(
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        bytes32 newContract
+    ) public {
+        vm.assume(action != 1);
+
+        bytes memory encodedUpgrade = abi.encodePacked(
+            module,
+            action,
+            chain,
+            newContract
+        );
+
+        assertEq(encodedUpgrade.length, 67);
+        
+        vm.expectRevert("invalid ContractUpgrade");
+
+        gs.parseContractUpgrade(encodedUpgrade);
+    }
+
+    function testParseContractUpgradeWrongAction_KEVM(
+        bytes32 module,
+        uint8 action,
+        uint16 chain,
+        bytes32 newContract
+    ) public symbolic(address(gs)) {
+        testParseContractUpgradeWrongAction(module, action, chain, newContract);
+    }
+
+    // Needs loop invariant for unbounded bytes type
+    function testContractUpgradeInvalidSize(bytes memory encodedUpgrade) public {
+        vm.assume(encodedUpgrade.length != 67);
+
+        vm.expectRevert();
+
+        gs.parseContractUpgrade(encodedUpgrade);
+    }
```

Comment:
We wrote these tests with symbolic execution using our KEVM tool in mind, and in that case this would not be an issue because, unlike fuzzing, symbolic execution covers every possible code path. However, we ended up not actually running symbolic execution on the `...InvalidSize` tests in this engagement (see my comment [here](https://github.com/wormhole-foundation/wormhole/pull/2956#discussion_r1240580012)), so based on your suggestions I have refactored the tests so they work better as fuzz tests.